### PR TITLE
 Fixed Routine setting  problem  investigation

### PR DESCRIPTION
**Pull request by spz11**
Fixed issue #<Mantis issue number>:16964: The survey settings are very slow to save, some even more than half a minute
New feature #<Mantis issue number>:bug/16964-survey-settings-slow-save
Dev:

### CODE REVIEW

**Comment by spz11** [comment]
When you save the general settings, you need to go through the  "$this->resetEM();" method to set the attributes of the questionnaire. When you set the language, do you just need to insert the corresponding extension table into the data

**Comment by spz11** [comment]
I have a questionnaire with more than 70 questions and 200 quotas. When adding other languages or not, it's very slow to click save

**Comment by olleharstedt** [comment]
Can you resolve the conflicts, please? They are related to PSR-12 compliance.

**Comment by Shnoulle** [comment]
Comment resetEM ???
https://github.com/LimeSurvey/LimeSurvey/pull/1716/files#diff-394de4d182b6812a0fd221499c29371dbd251634ae0b9f484a1e85ff07e34a1dR1190

Need a way to realky reset EM … else : if you do one survey, launch another survey : unsure result use new variable or old one …

File: application/helpers/common_helper.php [hunk 1]
```diff
@@ -3596,8 +3603,13 @@ function fixLanguageConsistency($sid, $availlangs = '')
     $query = "SELECT * FROM {{questions}} q JOIN {{question_l10ns}} ls ON ls.qid=q.qid WHERE sid='{$sid}'";
     $result = Yii::app()->db->createCommand($query)->query()->readAll();
     if (count($result) > 0) {
-        foreach ($result as $question) {
-            foreach ($langs as $lang) {
+        foreach ($langs as $lang) {
+            $query_count = "SELECT count(q.qid) FROM {{questions}} q JOIN {{question_l10ns}} ls ON ls.qid=q.qid WHERE sid='{$sid}' AND language='{$lang}'";
```
[the remark this answers is not in the document]
Comment (olleharstedt) [reply application/helpers/common_helper.php:3607]: Should also use query builder or model->find->count();

**Comment by c-schmitz** [comment]
Patch is stale and does not make sense in large parts.

**Pull request closed without merging**